ballerina/modules/parser/lexer.bal

```diff
@@ -103,22 +103,124 @@ public class Lexer {
         string word = "";
         Location location = self.currentLocation.clone();
         _ = self.readNextChar(); // Ignore first double quote character
-        while (true) {
+        string value = self.charReader.peek();
+        if value == DOUBLE_QUOTE {
+            value = self.readNextChar();
+            value = self.charReader.peek();
+            if value == DOUBLE_QUOTE {
+                word = self.readBlockStringLiteral();
+                return getToken(word.trim(), T_STRING, location);
+            } else {
+                return getToken(word, T_STRING, location);
+            }
+        } else {
+            while true {
+                value = self.charReader.peek();
+                if value is LineTerminator {
+                    string message = "Syntax Error: Unterminated string.";
+                    return error UnterminatedStringError(message, line = self.currentLocation.line,
+                                                         column = self.currentLocation.column);
+                } else {
+                    value = self.readNextChar();
+                    if value == DOUBLE_QUOTE && previousChar != BACK_SLASH {
+                        break;
+                    } else {
+                        word += value;
+                    }
+                    previousChar = value;
+                }
+            }
+            return getToken(word, T_STRING, location);
+        }
+    }
+
+    isolated function readBlockStringLiteral() returns string {
+        string previousChar = "";
+        string word = "";
+        string[] lines = [];
+        _ = self.readNextChar(); // Ignore third double quote character
+        while true {
             string value = self.charReader.peek();
-            if (value is LineTerminator) {
-                string message = "Syntax Error: Unterminated string.";
-                return error UnterminatedStringError(message, line = self.currentLocation.line,
-                                                     column = self.currentLocation.column);
-            } else if (value == DOUBLE_QUOTE && previousChar != BACK_SLASH) {
+            if value == NEW_LINE {
+                lines.push(word);
+                word = "";
                 value = self.readNextChar();
-                break;
+            } else if value == DOUBLE_QUOTE {
```

Comment:
if we use `else if value == DOUBLE_QUOTE && previousChar != BACK_SLASH {
`, we should use `else if value == DOUBLE_QUOTE && previousChar == BACK_SLASH {
` to remove the `\` in this special case `\" " "`. This may repeat the same thing.

Reply:
No, the point is to identify the end of the block string, which is `value == DOUBLE_QUOTE && previousChar != BACK_SLASH`, and then two consecutive `"` characters. Otherwise, it is just another character.